coordinator/core/clientapi.go

```diff
@@ -201,10 +178,36 @@ func (c *Core) UpdateManifest(ctx context.Context, rawUpdateManifest []byte) err
 	c.updateManifest = updateManifest
 	c.rawUpdateManifest = rawUpdateManifest
 	c.zaplogger.Info("An update manifest overriding package settings from the original manifest was set.")
-	_, err := c.sealState()
+
+	// Retrieve current recovery data before we seal the state again
```

Comment:
do this before setting `c.updateManifest` so that the state does not change on error